tests/parser/features/external_contracts/test_external_contract_calls.py

```diff
@@ -375,3 +377,130 @@ class Foo():
     def foo(arg2: num) -> num: pass
 """
     assert_tx_failed(lambda: get_contract(contract), exception = StructureException)
+
+
+def test_external_contract_call_declaration_expr():
+    contract_1 = """
+@public
+def bar() -> num:
+    return 1
+"""
+
+    contract_2 = """
+class Bar():
+    def bar() -> num: pass
+
+bar_contract: Bar
+
+@public
+def foo(contract_address: contract(Bar)) -> num:
+    self.bar_contract = contract_address
+    return self.bar_contract.bar()
+    """
+
+    c1 = get_contract(contract_1)
+    c2 = get_contract(contract_2)
+    assert c2.foo(c1.address) == 1
+
+
+def test_external_contract_call_declaration_stmt():
+    contract_1 = """
+lucky: num
+
+@public
+def set_lucky(_lucky: num):
+    self.lucky = _lucky
+
+@public
+def get_lucky() -> num:
+    return self.lucky
+"""
+
+    contract_2 = """
+class Bar():
+    def set_lucky(arg1: num): pass
+    def get_lucky() -> num: pass
+
+bar_contract: Bar
+
+@public
+def set_lucky(contract_address: contract(Bar)):
+    self.bar_contract = contract_address
+    self.bar_contract.set_lucky(1)
+
+@public
+def get_lucky(contract_address: contract(Bar)) -> num:
+    self.bar_contract = contract_address
+    return self.bar_contract.get_lucky()
```

Comment:
@DavidKnott I would love one more example/test here:

1.a) set the address of bar_contact to one contract (with something like, set_contract).
1.b) call get_lucky() and returns let's say 1.

2.a) then set the address to a secondary contract.
2.b) call get_luck() again an returns let's say 2.

It's basically just a slightly more elaborate version of this contract, but it steps over multiple blocks  ;)